### PR TITLE
[BugFix] Check sort key columns type when reorder primary key table

### PR DESCRIPTION
**Pull request by sevev**
Currently, the types of sort key columns are not fully supported, some types are not support(e.g. float/double), we will check the sort key columns type and reject create table request when we create a new table. However, no type checking is done when performing reorder operation. So a column of an unsupported type maybe defined as a sort key column after the reorder operation is executed and this will cause some unexpected error. (e.g. compaction task always failed)

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 4 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|0|4|00.00%|[603, 604, 605, 606]|

**Review by meegoo — DISMISSED** [review]

File: test/sql/test_ddl/R/test_alter_pk_reorder [hunk 1]
```diff
@@ -52,4 +52,38 @@ select * from tab1;
 500	500	500	500	500
 600	600	600	600	600
 700	700	700	700	700
+-- !result
```
Comment (chaoyli) [inline test/sql/test_ddl/R/test_alter_pk_reorder:55]: Five blank lines are too much.

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 2]
```diff
@@ -600,6 +600,11 @@ private void processReorderColumnOfPrimaryKey(ReorderColumnsClause alterClause,
             }
             int sortKeyIdx = targetIndexSchema.indexOf(oneCol.get());
             sortKeyIdxes.add(sortKeyIdx);
+            Type t = oneCol.get().getType();
+            if (!(t.isBoolean() || t.isIntegerType() || t.isLargeint() || t.isVarchar() || t.isDate() ||
+                    t.isDatetime())) {
+                throw new DdlException("Sort key column[" + colName + "] type not supported: " + t.toSql());
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:606]: `t.toSql()` is not necessary.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 4 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|0|4|00.00%|[603, 604, 605, 606]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 4 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|0|4|00.00%|[603, 604, 605, 606]|

**Review by decster — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**